rpc/lighthouse.go

```diff
@@ -52,55 +52,65 @@ func NewLighthouseClient(endpoint string, chainID *big.Int) (*LighthouseClient,
 }
 
 func (lc *LighthouseClient) GetNewBlockChan() chan *types.Block {
-	// setup health check & exit if the new block chan does no longer receive new blocks
+	// setup health check & reconnect if the new block chan no longer receive new blocks
 	lc.lastBlockSeenMux.Lock()
 	lc.lastBlockSeen = time.Now()
 	lc.lastBlockSeenMux.Unlock()
+	reConnect := make(chan int64, 10)
+	reConnectValue := int64(0)
 	go func() {
 		for ; ; time.Sleep(time.Second) {
 			lc.lastBlockSeenMux.Lock()
-			if time.Since(lc.lastBlockSeen) > time.Minute*2 {
-				lc.lastBlockSeenMux.Unlock()
-				// fatal if no new block was received for more than two minutes
-				logger.Fatalf("lighthouse client error, no new block retrieved since %v (%v ago)", lc.lastBlockSeen, time.Since(lc.lastBlockSeen))
+			if reConnectValue == 0 {
+				reConnectValue++
+				reConnect <- reConnectValue
+			} else if time.Since(lc.lastBlockSeen) > time.Minute*2 {
+				// if we did not receive a new block more than two minutes we try to reconnect to the lighthouse node
+				logger.Errorf("lighthouse client error, no new block retrieved since %v (%v ago)", lc.lastBlockSeen, time.Since(lc.lastBlockSeen))
+				lc.lastBlockSeen = time.Now()
+				reConnectValue++
+				reConnect <- reConnectValue
 			}
 			lc.lastBlockSeenMux.Unlock()
 		}
 	}()
```

Comment:
I don't quite understand why you have `reConnectValue` and why you do return an `int64` instead of a simple `bool`.

Why not simply do
```
reConnect := make(chan bool)
go func() {
	reConnect <- true

	for ; ; time.Sleep(time.Second) {
		lc.lastBlockSeenMux.Lock()
		if time.Since(lc.lastBlockSeen) > time.Minute*2 {
			// if we did not receive a new block more than two minutes we stry to reconnect to the lighthouse node
			logger.Errorf("lighthouse client error, no new block retrieved since %v (%v ago)", lc.lastBlockSeen, time.Since(lc.lastBlockSeen))
			lc.lastBlockSeen = time.Now()
			reConnect <- true
		}
		lc.lastBlockSeenMux.Unlock()
	}
}()
```

:exclamation: Please also note that the `reConnect` channel is not buffered anymore. This means that the initial `reConnect` command blocks until it is received by the go function below.

Why has it been buffered before?

Reply:
its now bool